### DISCUSSION TITLE
Regarding this project( distributed computing)

### DISCUSSION THREAD

**Discussion by kae-at-seventh-root** [BOINC/boinc#5586] [Q&A]
Hi there,
I'm interested in this project (distributed computing) and have some questions. I'm particularly interested in this project because it allows smartphones to participate in computations. However, seeing that CGI is used on the server side makes me think that it might be using somewhat outdated technology. Therefore, I have a question: Are there other frameworks or projects that allow the use of smartphones (iPhone, Android) as worker nodes? I hope it's not rude to ask this question here, but I don't have much knowledge on the subject, so I wanted to ask.

Thank you.

**Comment by AenBleidd** [comment]
@kae-at-seventh-root, we prefer to use the technologies that fits the best their purpose.
And we are using updated technologies on both server and client side.
For example, we have (mostly) rewritten the Java code for Android BOINC to Kotlin.
On the desktop, we are using c++ with the up-to-date libraries in our clients.
So, I'd suggest to give BOINC a try.

If you're looking for other volunteer project (and really afraid of the CGI on the server side), you could try DreamLab, that is available for Android and iPhone: https://www.vodafone.com/vodafone-foundation/focus-areas/dreamlab-app

**Comment by kae-at-seventh-root** [comment]
@AenBleidd 
Thank you for your answer.
I did not know that the code had been updated. There was a lot of activity on GitHub PRs, so I was hopeful. For now, I think I will try running the server on my local machine using something like docker-compose. If I'm building an environment for a MasterNode, is this the right place to look? https://github.com/BOINC/boinc/wiki/Create-a-BOINC-server-(cookbook)

Where can I find information on setting up a WorkerNode environment?

**Comment by AenBleidd** [comment]
@kae-at-seventh-root, why do you need to run the server? Do you have in mind a Project you want to run?

**Reply by kae-at-seventh-root** [reply]
@AenBleidd 
Yes. I'm interested in distributed computing, so I was curious about how the server operates. What do you mean by "the project you want to execute"?

**Reply by AenBleidd** [reply]
Assuming this question, you don't need a server, you need to download client (and optionally Manager) and connect to the existing projects. See this manual for details: https://boinc.berkeley.edu/wiki/User_manual

**Reply by kae-at-seventh-root** [reply]
Sure.
Thank you so much.
